src/Password/Argon2i.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-crypt for the canonical source repository
+ * @copyright Copyright (c) 2005-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-crypt/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Crypt\Password;
+
+use Traversable;
+use Zend\Stdlib\ArrayUtils;
+
+use const PASSWORD_ARGON2I;
+use const PHP_VERSION_ID;
+
+use function is_array;
+use function password_hash;
+use function password_verify;
+use function strtolower;
+
+/**
+ * Argon2i algorithm using password_hash() function of PHP
+ */
+class Argon2i implements PasswordInterface
+{
+    /**
+     * Maximum memory (in kibibytes) that may be used to compute the Argon2 hash
+     *
+     * @var int|null
+     */
+    protected $memoryCost;
```

Comment:
I will suggest to assign the default PHP values for `$memoryCost`, `$timeCost` and `$threads`. Using the constants:
```php
PASSWORD_ARGON2_DEFAULT_MEMORY_COST
PASSWORD_ARGON2_DEFAULT_TIME_COST
PASSWORD_ARGON2_DEFAULT_THREADS
```

Reply:
Using the constants on PHP < 7.2 leads to an "Undefined constant" error even if you just autoload the class. (See the <7.2 Unit-Tests).